<launch>
  <!-- We resume the logic in empty_world.launch, changing only the name of the world to be launched -->
	<include file="$(find gazebo_ros)/launch/empty_world.launch">
		<arg name="world_name" value="$(find hku_m100_gazebo)/worlds/sphere_world.world"/>
		<!-- more default parameters can be changed here -->
		<arg name="verbose" value="true"/>
		<arg name="paused" value="false"/>
		<arg name="use_sim_time" value="true"/>
		<arg name="gui" value="true"/>
		<arg name="recording" value="false"/>
		<arg name="debug" value="false"/>
	</include>


	

	<rosparam param="nr_of_obstacles">100</rosparam>
		<!-- Spawn simulated quadrotor uav -->
	<include file="$(find hku_m100_gazebo)/launch/spawn_hku_m100.launch" >
    	<arg name="model" value="$(find hku_m100_gazebo)/urdf/hku_m100.urdf.xacro"/> 
	</include>
  

	<node pkg="hku_m100_gazebo" type="rviz_collision_publisher" name="rviz_collision_publisher" output="screen"/>
  <!-- Start rviz visualization with preset config -->
  <!--<node pkg="rviz" type="rviz" name="rviz" args="-d $(find hector_quadrotor_demo)/rviz_cfg/outdoor_flight.rviz"/>-->
	<node name="rviz" pkg="rviz" type="rviz" args="-d $(find hku_m100_gazebo)/rviz_cfg/flight.rviz"/>

</launch>
